snuba/web/db_query.py

```diff
@@ -73,117 +221,39 @@ def raw_query(
     TODO: As soon as we have a StorageQuery abstraction remove all the references
     to the original query from the request.
     """
-
-    use_cache, use_deduper, uc_max = state.get_configs(
-        [
-            ("use_cache", settings.USE_RESULT_CACHE),
-            ("use_deduper", 1),
-            ("uncompressed_cache_max_cols", 5),
-        ]
-    )
-
     all_confs = state.get_all_configs()
     query_settings: MutableMapping[str, Any] = {
         k.split("/", 1)[1]: v
         for k, v in all_confs.items()
         if k.startswith("query_settings/")
     }
 
-    # Experiment, if we are going to grab more than X columns worth of data,
-    # don't use uncompressed_cache in clickhouse, or result cache in snuba.
-    if len(request.query.get_all_referenced_columns()) > uc_max:
-        query_settings["use_uncompressed_cache"] = 0
-        use_cache = 0
-
     timer.mark("get_configs")
 
     sql = query.format_sql()
-    query_id = md5(force_bytes(sql)).hexdigest()
-    with state.deduper(query_id if use_deduper else None) as is_dupe:
-        timer.mark("dedupe_wait")
 
-        result = cache.get(query_id) if use_cache else None
-        timer.mark("cache_get")
+    update_with_status = partial(
+        update_query_metadata_and_stats,
+        request,
+        sql,
+        timer,
+        stats,
+        query_metadata,
+        query_settings,
+        trace_id,
+    )
 
-        stats.update(
-            {
-                "is_duplicate": is_dupe,
-                "query_id": query_id,
-                "use_cache": bool(use_cache),
-                "cache_hit": bool(result),
-            }
-        ),
-
-        update_with_status = partial(
-            update_query_metadata_and_stats,
-            request,
-            sql,
-            timer,
-            stats,
-            query_metadata,
-            query_settings,
-            trace_id,
+    try:
+        result = execute_query_with_deduplication(
+            request, query, timer, stats, query_settings,
         )
-
-        if not result:
-            try:
```

Comment:
Note that the scope of this `try` block has changed, so it will now also catch exceptions thrown by the deduplication or cache steps. I don't see why this would be a problem, but it is a different behavior than what currently exists.